iup/eval_projections.py

```diff
@@ -1,5 +1,86 @@
+import polars as pl
 import altair as alt
 
+# get mspe #
+def get_mspe(
+        data_df,
+        pred_df,
+        var
+):
+    """
+    Calculate the mean squared prediction error between the observed
+    data and prediction when both are available. 
+
+    Parameters:
+    --------------
+    data_df: polars.DataFrame
+        The observed vaccine uptake. Must include `date` and the variable-
+        of-interest (var).
+    pred_df: polars.DataFrame
+        The predicted daily uptake. Must include `date` and the variable-
+        of-interest (var).
+    var: string
+        The name of the variable-of-interest.
+    
+    Return:
+    -------------
+    A polars.DataFrame with the MSPE and the date of initializing forecast. 
+    
+    """
+    
+    
+    mspe = data_df.join(
+            pred_df, 
+            on = 'date',
+            how = 'inner'
```

Comment:
this is a good place for validation. I assume you want 1:1 matching?

Reply:
Yes. The rows that has the mutual `date` between `data_df` and `pred_df` will be selected.